superset/assets/javascripts/SqlLab/components/SqlEditor.jsx

```diff
@@ -76,7 +76,9 @@ class SqlEditor extends React.PureComponent {
     if (!this.props.database.allow_run_sync) {
       effectiveRunAsync = true;
     }
-    this.startQuery(effectiveRunAsync);
+    if (this.props.queryEditor.sql) {
```

Comment:
If that's the right place to add it, maybe an early return at the top of the function instead? 

```
if (!this.props.queryEditor.sql)
    return;
```